spring-core/src/main/java/org/springframework/core/io/support/PropertySourceProcessor.java

```diff
@@ -80,7 +80,7 @@ public void processPropertySource(PropertySourceDescriptor descriptor) throws IO
 		String name = descriptor.name();
 		String encoding = descriptor.encoding();
 		List<String> locations = descriptor.locations();
-		Assert.isTrue(locations.size() > 0, "At least one @PropertySource(value) location is required");
+		Assert.notEmpty(locations, "At least one @PropertySource(value) location is required");
```

Comment:
The `locations()` property of the `PropertySourceDescriptor` record is not declared as `@Nullable`.

See previous comment for further details on nullability constraints.